### PR TITLE
LPS-93141 Add tests for data definition rule functions

### PR DESCRIPTION
**Pull request by mclomello**
@jeyvison sended again as we talked

### CODE REVIEW

**Comment by mclomello** [comment]
ci:test:sf

**Comment by mclomello** [comment]
ci:test:relevant

File: modules/apps/data-engine/data-engine-rest-impl/build.gradle [hunk 1]
```diff
@@ -28,4 +28,7 @@ dependencies {
 	compileOnly project(":apps:portal-vulcan:portal-vulcan-api")
 	compileOnly project(":core:petra:petra-function")
 	compileOnly project(":core:petra:petra-string")
+
+	testCompile project(":apps:data-engine:data-engine-rest-client")
```
Comment (jeyvison) [inline modules/apps/data-engine/data-engine-rest-impl/build.gradle]: Remove this

File: modules/apps/data-engine/data-engine-rest-impl/src/test/java/com/liferay/data/engine/rest/internal/rule/function/v1_0/DataDefinitionRulesEmptyResourceTest.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+			DataDefinitionFieldUtil.toSPIDataDefinitionField(
+				dataDefinitionField),
+			_dataRecord.getDataRecordValues(
+			).get(
+				dataDefinitionField.getName()
+			));
+	}
+
+	private DataDefinitionRule[] _dataDefinitionEmptyRules() {
+		return new DataDefinitionRule[] {
```
Comment (jeyvison) [inline modules/apps/data-engine/data-engine-rest-impl/src/test/java/com/liferay/data/engine/rest/internal/rule/function/v1_0/DataDefinitionRulesEmptyResourceTest.java]: we should not return an array if we are using only one value.

File: modules/apps/data-engine/data-engine-rest-impl/src/test/java/com/liferay/data/engine/rest/internal/rule/function/v1_0/DataDefinitionRulesIntegerLiteralResourceTest.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+	protected DataRuleFunctionResult getDataRuleFunctionResult() {
+		DataDefinitionRule dataDefinitionRule =
+			_dataDefinitionIntegerLiteralRules()[0];
+
+		DataRuleFunction dataRuleFunction =
+			new IntegerLiteralDataRuleFunction();
+
+		DataDefinitionField dataDefinitionField =
+			randomDataDefinitionFields("age", "numeric")[0];
```
Comment (jeyvison) [inline modules/apps/data-engine/data-engine-rest-impl/src/test/java/com/liferay/data/engine/rest/internal/rule/function/v1_0/DataDefinitionRulesIntegerLiteralResourceTest.java]: We should not return and array of we're always getting the first value.

File: modules/apps/data-engine/data-engine-rest-impl/src/test/java/com/liferay/data/engine/rest/internal/rule/function/v1_0/DataDefinitionRulesMatchExpressionResourceTest.java [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+/**
+ * @author Marcelo Mello
+ */
+public class DataDefinitionRulesMatchExpressionResourceTest
+	extends BaseDataDefinitionRulesTestCase {
+
+	@Before
+	public void setUp() {
+		_dataDefinition = new DataDefinition();
```
Comment (jeyvison) [inline modules/apps/data-engine/data-engine-rest-impl/src/test/java/com/liferay/data/engine/rest/internal/rule/function/v1_0/DataDefinitionRulesMatchExpressionResourceTest.java]: We dont need this dataDefinition. Create a dataDefinitionRule directly.

**Pull request closed without merging**